pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -145,7 +145,7 @@ func (r *AtlasDataFederationReconciler) Reconcile(context context.Context, req c
 		return r.handleDelete(ctx, log, dataFederation, project, dataFederationService).ReconcileResult(), nil
 	}
 
-	err = customresource.ApplyLastConfigApplied(context, project, r.Client)
+	err = customresource.ApplyLastConfigApplied(context, dataFederation, r.Client)
```

Comment:
Seems like another bug fixed here right?

Reply:
yes, it's part of the ticket being solved.